clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoFilesetCatalog.java

```diff
@@ -0,0 +1,228 @@
+import com.datastrato.gravitino.dto.requests.FilesetCreateRequest;
+import com.datastrato.gravitino.dto.requests.FilesetUpdateRequest;
+import com.datastrato.gravitino.dto.requests.FilesetUpdatesRequest;
+import com.datastrato.gravitino.dto.responses.DropResponse;
+import com.datastrato.gravitino.dto.responses.EntityListResponse;
+import com.datastrato.gravitino.dto.responses.FilesetResponse;
+import com.datastrato.gravitino.exceptions.FilesetAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.NoSuchFilesetException;
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import com.datastrato.gravitino.file.Fileset;
+import com.datastrato.gravitino.file.FilesetCatalog;
+import com.datastrato.gravitino.file.FilesetChange;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Fileset catalog is a catalog implementation that supports fileset like metadata operations, for
+ * example, schemas and filesets list, creation, update and deletion. A Fileset catalog is under the
+ * metalake.
+ */
+public class GravitinoFilesetCatalog extends GravitinoCatalog implements FilesetCatalog {
+
+  GravitinoFilesetCatalog(
+      String name,
+      Type type,
+      String provider,
+      String comment,
+      Map<String, String> properties,
+      AuditDTO auditDTO,
+      RESTClient restClient) {
+    super(name, type, provider, comment, properties, auditDTO, restClient);
+  }
+
+  @Override
+  public FilesetCatalog asFilesetCatalog() throws UnsupportedOperationException {
+    return this;
+  }
+
+  /**
+   * List the filesets in a schema namespace from the catalog.
+   *
+   * @param namespace A schema namespace.
+   * @return An array of fileset identifiers in the namespace.
+   * @throws NoSuchSchemaException If the schema does not exist.
+   */
+  @Override
+  public NameIdentifier[] listFilesets(Namespace namespace) throws NoSuchSchemaException {
+    Namespace.checkFileset(namespace);
+
+    EntityListResponse resp =
+        restClient.get(
+            formatFilesetRequestPath(namespace),
+            EntityListResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.filesetErrorHandler());
+    resp.validate();
+
+    return resp.identifiers();
+  }
+
+  /**
+   * Load fileset metadata by {@link NameIdentifier} from the catalog.
+   *
+   * @param ident A fileset identifier.
+   * @return The fileset metadata.
+   * @throws NoSuchFilesetException If the fileset does not exist.
+   */
+  @Override
+  public Fileset loadFileset(NameIdentifier ident) throws NoSuchFilesetException {
+    NameIdentifier.checkFileset(ident);
+
+    FilesetResponse resp =
+        restClient.get(
+            formatFilesetRequestPath(ident.namespace()) + "/" + ident.name(),
```

Comment:
Do you need to encode the `ident.name()`, in case there are some special character

Reply:
@mchades Should we validate the special character to avoid using a special character? The `fileset` is managed by gravitino.

Reply:
I think it is difficult to enumerate the special characters for validating, using `URLEncoder`  is enough here.